contracts/lockdrop/src/contract.rs

```diff
@@ -392,8 +455,427 @@ pub fn query(deps: Deps, env: Env, msg: QueryMsg) -> StdResult<Binary> {
+                    pool_info.lp_token = new_pool_info.lp_token;
+                    ASSET_POOLS.save(deps.storage, *pool_type, &pool_info, env.block.height)?;
+                }
+                MIGRATION_STATUS.save(deps.storage, &MigrationState::Completed)?;
+                attrs.push(attr("migration_completed", "true"));
+            } else {
+                attrs.push(attr("users_count", users.len().to_string()));
+                //iterate over users
+                for user in users {
+                    for pool_type in &pool_types {
+                        let mut total_lokups = Uint128::zero();
+                        let lookup_infos: Vec<(u64, LockupInfoV2)> = LOCKUP_INFO
+                            .prefix((*pool_type, &user))
+                            .range(deps.storage, None, None, Order::Ascending)
+                            .collect::<Result<Vec<(u64, LockupInfoV2)>, StdError>>()?;
+                        for (duration, mut lockup_info) in lookup_infos {
+                            let info = query_lockup_info(
+                                deps.as_ref(),
+                                &env.clone(),
+                                user.as_ref(),
+                                *pool_type,
+                                duration,
+                            )?;
+                            let p: String = (*pool_type).into();
+                            let kf = kfs
+                                .get(&p)
+                                .ok_or_else(|| StdError::generic_err("Can't get kf"))?;
+                            lockup_info.generator_proxy_debt = info.generator_proxy_debt;
+                            lockup_info.lp_units_locked =
+                                (*kf).checked_mul_uint256(lockup_info.lp_units_locked.into())?;
+                            LOCKUP_INFO.save(
+                                deps.storage,
+                                (*pool_type, &user, duration),
+                                &lockup_info,
+                            )?;
+                            total_lokups += lockup_info.lp_units_locked;
+                        }
+                        TOTAL_USER_LOCKUP_AMOUNT.update(
+                            deps.storage,
+                            (*pool_type, &user),
+                            env.block.height,
+                            |lockup_amount| -> StdResult<Uint128> {
+                                if let Some(la) = lockup_amount {
+                                    Ok(la.checked_add(total_lokups)?)
+                                } else {
+                                    Ok(total_lokups)
+                                }
+                            },
+                        )?;
+                    }
+                }
+                MIGRATION_USERS_COUNTER.save(deps.storage, &(current_skip + limit as u32))?;
+                MIGRATION_STATUS.save(deps.storage, &MigrationState::MigrateUsers(page + 1u64))?;
```

Comment:
what's the point of storing the page?

Reply:
it stores migrated count, not the page. 
reason: what if you change count of migrating users on the go?